tests/test.cpp

```diff
@@ -537,6 +537,57 @@ void TriviallyCopyableTest() {
   #endif
 }
 
+// Check stringify of an default enum value to json
+void JsonDefaultEnumTest() {
+  flatbuffers::Parser parser;
+  const char *test_table = 
+    "enum Color: byte { Red=0, Green=1, Blue=2 }\n"
+    "table Test{\n"
+    "  d : Color;\n"
+    "  r : Color = Red;\n"
+    "  g : Color = Green;\n"
+    "  b : Color = Blue;\n"
+    "  x : float = 3.14;\n"
+    "}\n"
+    "root_type Test;\n"
+    "file_identifier \"TEST\";";
+  bool parser_done = parser.Parse(test_table);
+  TEST_EQ(parser_done, true);
+
+  std::string jsongen;
+  flatbuffers::FlatBufferBuilder fbb;
+  parser.opts.output_enum_identifiers = true;
+  parser.opts.output_default_scalars_in_json = true;
+  // test#1: empty table
+  {
+    fbb.Clear();
+    jsongen.clear();
+    auto start = fbb.StartTable();
+    const auto end = fbb.EndTable(start);
+    auto root = flatbuffers::Offset<Monster>(end);
+    fbb.Finish(root, "TEST");
+    GenerateText(parser, fbb.GetBufferPointer(), &jsongen);
+    std::string test1_expectd = "{\n  d: Red,\n  r: Red,\n  g: Green,\n  b: Blue,\n  x: 3.14\n}\n";
+    TEST_EQ(jsongen, test1_expectd);
+  }
+  // test#2: explicit initialisation test
+  {
+    fbb.Clear();
+    jsongen.clear();
+    auto start = fbb.StartTable();
+    fbb.AddElement<int8_t>(4, 2, 0); // set Test.d = Blue, default is Red
```

Comment:
Probably better to remove this test.. users shouldn't be calling `AddElement` directly, or use hard-coded offsets.